src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -312,47 +315,19 @@ private void RefreshCondensedDisplay(Action refreshDisplay)
                 return;
             }
 
-            string newContent = "null";
+            CompactBubbleViewModel newContent = null;
 
             RunOnSchedulerSync(
                 () =>
                 {
                     var mirrorData = nodeViewModel.NodeModel.CachedValue;
-                    if (mirrorData != null)
-                    {
-                        if (mirrorData.IsCollection)
-                        {
-                            // TODO(Ben): Can we display details of the array and 
-                            // probably display the first element of the array (even 
-                            // when it is multi-dimensional array)?
-                            newContent = Wpf.Properties.Resources.PreviewListLabel;
-                        }
-                        else if (mirrorData.Data == null && !mirrorData.IsNull && mirrorData.Class != null)
-                        {
-                            newContent = mirrorData.Class.ClassName;
-                        }
-                        else if (mirrorData.Data is Enum)
-                        {
-                            newContent = ((Enum)mirrorData.Data).GetDescription();
-                        }
-                        else
-                        {
-                            if (String.IsNullOrEmpty(mirrorData.StringData))
-                            {
-                                newContent = String.Empty;
-                                return;
-                            }
-
-                            int index = mirrorData.StringData.IndexOf('(');
-                            newContent = index != -1 ? mirrorData.StringData.Substring(0, index) : mirrorData.StringData;
-                        }
-                    }
-                },
+                    newContent = CompactBubbleHandler.Process(mirrorData);                },
```

Comment:
I think the handler is fine, just curious why this logic can't just live in the CompactPreviewViewModels constructors.